chains/solana/contracts/programs/example-ccip-sender/Cargo.toml

```diff
@@ -13,10 +13,12 @@ no-entrypoint = []
 no-idl = []
 no-log-ix-name = []
 cpi = ["no-entrypoint"]
-default = []
+default = ["custom-heap"]
+custom-heap = []
 
 [dependencies]
 solana-program = "1.17.25" # pin solana to 1.17
 anchor-lang = { version = "0.29.0", features = [] }
 anchor-spl = "0.29.0"
 fee_quoter = { version = "0.1.0-dev", path = "../fee-quoter", features = ["no-entrypoint"]}
+smalloc = "0.1.2"
```

Comment:
I'm reviewing this allocator's internals. In the meantime however, have you checked if there is any allocator that is more commonly used in the community? This repo doesn't seem to have many eyes on it...
https://github.com/oxfeeefeee/smalloc

I may even prefer the more-trivial approach of https://solana.stackexchange.com/questions/7311/handling-out-of-memory-issues-in-heap personally. Any thoughts? Maybe @archseer also wants to chime in here?